APA PsycNET.js

```diff
@@ -1,162 +1,199 @@
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
 	}
+	return found ? items : false;
 }
 
-//default field xpath
-var fields = {
-	title: '(//div[@id="rdcTitle"])[1]',
-	authors: '(//div[@id="rdcAuthors"])[1]',
-	voliss: '(//div[@id="rdcSource"])[1]',
-	abstract: '//div[@id="rdRecord"]/div[@class="rdRecordSection"][2]'
-}
 
-function getField(field, doc) {
-	var val = ZU.xpathText(doc, field);
-	if(val) val = ZU.trimInternal(val);
-	return val;
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
 }
 
-//for scraping publication information directly from pages
-var volissRe = {
-	journalArticle: 
-		/^(.+?)(?:,\sVol\s(\d+)\((\d+)\))?,\s(\w+\s(?:\d+\s*,\s)?\d{4}),\s(?:(\d+-\d+)|No Pagination Specified).(?:\sdoi:\s(.+))?$/i,
-	bookSection:
-		/^(.+?),\s\((\d{4})\)\.\s(.+?),\s\(pp\.\s(\d+-\d+)\)\.\s(.+?):\s(.+?),\s(?:(\w+))?,\s(\d+)\spp\.(?:\sdoi:\s(.+))?/i,
-	book:
-		/^(.+?):\s(.+?)(?:\.\s\((\d{4})\)\.\s(\w+)\s(\d+)\spp\.\sdoi:\s(.+))?$/i
-};
 
-var creatorMap = {
-	Ed: 'editor'
-};
-
-function doWeb(doc, url) {
-	var type = detectWeb(doc, url);
-	if (type == "multiple") {
-		var items = new Object();
-		var titles = ZU.xpath(doc, '//div[@class="srhcTitle"]/a');
-		for(var i=0, n=titles.length; i<n; i++) {
-			items[titles[i].href] = titles[i].textContent;
+function scrape(doc, url) {
+	var uid = getIds(doc,url);
+	
+	var productCode;
+	var db = doc.getElementById('database');
+	if (db) {
+		var db = db.parentNode.textContent;
+		if (db.indexOf('PsycARTICLES')>-1) {
+			productCode = 'PA';
+		} else if (db.indexOf('PsycBOOKS')>-1) {
+			productCode = 'PB';
+		} else if (db.indexOf('PsycINFO')>-1) {
+			productCode = 'PI';
 		}
-		Zotero.selectItems(items, function(selectedItems) {
-			if(!selectedItems) return true;
+	}
+	Z.debug(productCode);
 	
-			var arts = new Array();
-			for(var i in selectedItems) {
-				arts.push(i);
-			}
-			ZU.processDocuments(arts, scrape);
+	var postData = '{"api":"record.exportRISFile","params":{"UIDList":[{"UID":"'+uid+'","ProductCode":"'+productCode+'"}],"exportType":"zotero"}}';
+	var headers = {
+		'Content-Type': 'application/json',
+		'Referer': url
+	};
+	Z.debug(postData);
+
+
+	ZU.doPost('/api/request/record.exportRISFile', postData, function(text) {
+		try {
+			var data = JSON.parse(text);
+		} catch(e) {
+			Z.debug('POST request did not result in valid JSON');
+			Z.debug(text);
+		};
+Z.debug(text)
+		if (data && data.isRisExportCreated) {
+			Z.debug(data);
+			ZU.doGet('/ris/download', function(text2) {
+				Z.debug("TEXT 2: "+text2);
+				var redirect = text2.match(/<meta http-equiv="refresh" content="10; url=([^"]*)"/)
```

Comment:
Okay, I can now replicate this: If I make more than three tests in Scaffold then this redirect to a captcha query page is appearing:
```html
         <html>
         
         <head>
         <META NAME="ROBOTS" CONTENT="NOINDEX, NOFOLLOW">
         <meta http-equiv="cache-control" content="max-age=0" />
         <meta http-equiv="cache-control" content="no-cache" />
         <meta http-equiv="expires" content="0" />
         <meta http-equiv="expires" content="Tue, 01 Jan 1980 1:00:00 GMT" />
         <meta http-equiv="pragma" content="no-cache" />
         <meta http-equiv="refresh" content="10; url=/distil_r_captcha.html?requestId=8858330d-2a2e-416a-9372-bc4f9cb41117&httpReferrer=%2Fris%2Fdownload" />
         <script type="text/javascript">
         	(function(window){
         		try {
         			if (typeof sessionStorage !== 'undefined'){
         				sessionStorage.setItem('distil_referrer', document.referrer);
         			}
         		} catch (e){}
         	})(window);
         </script>
         <script type="text/javascript" src="/ga849264.js" defer></script><style type="text/css">#d__fFH{position:absolute;top:-5000px;left:-5000px}#d__fF{font-family:serif;font-size:200px;visibility:hidden}#dydzyfdedwfbyeetb{display:none!important}</style></head>
         <body>
         <div id="distilIdentificationBlock">&nbsp;</div>
         </body>
         </html>
```
I cannot solve the captcha within Zotero, thus currently I have to reload the url in the browser tab of Scaffold (to trigger whatever is needed). I have to test this more, if this is only an issue for tests.

@dstillman Could this `sessionStorage` be important to avoid being classified as a bot?